src/net_processing.cpp

```diff
@@ -1665,17 +1680,19 @@ bool static AlreadyHaveBlock(const uint256& block_hash) EXCLUSIVE_LOCKS_REQUIRED
     return LookupBlockIndex(block_hash) != nullptr;
 }
 
-void RelayTransaction(const uint256& txid, const uint256& wtxid, const CConnman& connman)
+void RelayTransaction(const uint256& txid, const uint256& wtxid, const CConnman& connman, bool flood)
 {
-    connman.ForEachNode([&txid, &wtxid](CNode* pnode) EXCLUSIVE_LOCKS_REQUIRED(::cs_main) {
+    connman.ForEachNode([&txid, &wtxid, flood](CNode* pnode) EXCLUSIVE_LOCKS_REQUIRED(::cs_main) {
         AssertLockHeld(::cs_main);
 
         CNodeState* state = State(pnode->GetId());
         if (state == nullptr) return;
         if (state->m_wtxid_relay) {
-            pnode->PushTxInventory(wtxid);
+            pnode->PushTxInventory(wtxid, flood);
         } else {
-            pnode->PushTxInventory(txid);
+            // Reconciliations are not supported for non-wtxid peers,
+            // so we always use flooding.
+            pnode->PushTxInventory(txid, true);
```

Comment:
I know it's more complicated logic, but think it's necessary.
m_flood inside peer's ReconState is one factor, but it's not sufficient, because we can't flood *all* transactions to them.

If we did so, there will be too much flooding. A non-reachable node always sets m_flood for its 8 outbound peers to true. In that case, it will flood all transactions it gets for them, too much bandwidth.

I suggest flooding only transactions received via inbound flooding, so this is never the case for non-reachable nodes.

I'm open to suggestions on other ways how to limit flooding, but I think this one makes the most sense.